src/components/Editor/index.js

```diff
@@ -585,48 +573,36 @@ class Editor extends PureComponent<Props, State> {
   }
 }
 
-Editor.propTypes = {
-  hitCount: PropTypes.object,
-  selectedLocation: PropTypes.object,
-  selectedSource: ImPropTypes.map,
-  searchOn: PropTypes.bool,
-  addOrToggleDisabledBreakpoint: PropTypes.func,
-  toggleBreakpoint: PropTypes.func,
-  selectSource: PropTypes.func,
-  jumpToMappedLocation: PropTypes.func,
-  coverageOn: PropTypes.bool,
-  selectedFrame: PropTypes.object,
-  searchModifiers: PropTypes.object,
-  query: PropTypes.string,
-  horizontal: PropTypes.bool,
-  startPanelSize: PropTypes.number,
-  endPanelSize: PropTypes.number,
-  conditionalPanelLine: PropTypes.number,
-  openConditionalPanel: PropTypes.func,
-  closeConditionalPanel: PropTypes.func,
-  continueToHere: PropTypes.func,
-  setContextMenu: PropTypes.func
-};
-
 Editor.contextTypes = {
   shortcuts: PropTypes.object
 };
 
-export default connect(
-  state => {
-    const selectedSource = getSelectedSource(state);
-    const sourceId = selectedSource ? selectedSource.get("id") : "";
-    return {
-      selectedLocation: getSelectedLocation(state),
-      selectedSource,
-      searchOn: getActiveSearch(state) === "file",
-      hitCount: getHitCountForSource(state, sourceId),
-      selectedFrame: getSelectedFrame(state),
-      query: getFileSearchQuery(state),
-      modifiers: getFileSearchModifiers(state),
-      coverageOn: getCoverageEnabled(state),
-      conditionalPanelLine: getConditionalPanelLine(state)
-    };
-  },
-  dispatch => bindActionCreators(actions, dispatch)
-)(Editor);
+const mapStateToProps = state => {
+  const selectedSource = getSelectedSource(state);
+  const sourceId = selectedSource ? selectedSource.get("id") : "";
+  return {
+    selectedLocation: getSelectedLocation(state),
+    selectedSource,
+    searchOn: getActiveSearch(state) === "file",
+    hitCount: getHitCountForSource(state, sourceId),
+    selectedFrame: getSelectedFrame(state),
+    coverageOn: getCoverageEnabled(state),
+    conditionalPanelLine: getConditionalPanelLine(state)
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    openConditionalPanel: line => dispatch(openConditionalPanel(line)),
+    closeConditionalPanel: () => dispatch(closeConditionalPanel()),
+    setContextMenu: (type, event) => dispatch(setContextMenu(type, event)),
+    continueToHere: line => dispatch(continueToHere(line)),
+    jumpToMappedLocation: location => dispatch(jumpToMappedLocation(location)),
+    toggleBreakpoint: line => dispatch(toggleBreakpoint(line)),
+    addOrToggleDisabledBreakpoint: line =>
+      dispatch(addOrToggleDisabledBreakpoint(line)),
+    traverseResults: (rev, editor) => dispatch(traverseResults(rev, editor))
+  };
```

Comment:
Ooh, I like this style, but let's discuss adopting it in a separate PR. 

I honk we can land his fix without I and see what style for actions is most used in the community. I would Perper something as explicit, but more terse. It might not exis :)

Reply:
I can roll back the style changes. Let's have the discussion here: https://github.com/devtools-html/debugger.html/pull/4454

Reply:
Just so this doesn't get lost, we can avoid the function entirely like [this](https://github.com/devtools-html/debugger.html/pull/4454#discussion_r146109706).